flake.nix

```diff
@@ -1,98 +1,30 @@
 {
   description = "agenix-cli";
 
-  inputs.nixpkgs.url = "github:nixos/nixpkgs/nixos-21.11";
-  inputs.crate2nix = { url = "github:kolloch/crate2nix"; flake = false; };
-  inputs.mozilla = { url = "github:mozilla/nixpkgs-mozilla"; flake = false; };
+  inputs.nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+  inputs.flake-utils.url = "github:numtide/flake-utils";
 
-  outputs =
-    { self
-    , nixpkgs
-    , mozilla
-    , ...
-    } @ inputs:
+  outputs = { self, nixpkgs, flake-utils }: flake-utils.lib.eachDefaultSystem (system:
     let
-      nameValuePair = name: value: { inherit name value; };
-      genAttrs = names: f: builtins.listToAttrs (map (n: nameValuePair n (f n)) names);
-      allSystems = [ "x86_64-linux" "aarch64-linux" "i686-linux" "x86_64-darwin" ];
-
-      # rustOverlay = final: prev:
-      #   let
-      #     rustChannel = prev.rustChannelOf {
-      #       channel = "1.47.0";
-      #       sha256 = "1hkisci4as93hx8ybf13bmxkj9jsvd4a9ilvjmw6n64w4jkc1nk9";
-      #     };
-      #   in
-      #   {
-      #     inherit rustChannel;
-      #     rustc = rustChannel.rust;
-      #     cargo = rustChannel.rust;
-      #   };
-
-      forAllSystems = f: genAttrs allSystems (system: f {
-        inherit system;
-        pkgs = import nixpkgs {
-          inherit system;
-          # overlays = [
-          #   (import "${mozilla}/rust-overlay.nix")
-          #   rustOverlay
-          # ];
-        };
-      });
-
-      crate2nix = forAllSystems ({ system, ... }:
-        let
-          sources = import "${inputs.crate2nix}/nix/sources.nix";
-          pkgs = import sources.nixpkgs { inherit system; };
-          cargoNix = import "${inputs.crate2nix}/crate2nix/Cargo.nix" {
-            inherit pkgs;
-            defaultCrateOverrides = pkgs.defaultCrateOverrides // {
-              cssparser-macros = { ... }: {
-                buildInputs = with pkgs;
-                  lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.Security ];
-              };
-            };
-          };
-        in
-        cargoNix.rootCrate.build);
+      pkgs = nixpkgs.legacyPackages.${system};
     in
     {
-      # for use with update.sh script
-      inputs = builtins.removeAttrs inputs [ "self" ];
+      packages.agenix-cli = pkgs.callPackage
```

Comment:
Why does this need to be `callPackage`d? Could it not just use `rustPlatform.buildRustPackage` directly?

Reply:
I... did not realize this, I guess I've been doing it wrong this entire time. 😅